src/lib.rs

```diff
@@ -1,27 +1,41 @@
 use std::error::Error;
 
 mod cpu;
+mod controller;
+
 use cpu::Cpu;
+use crate::controller::{Controller, ControllerState};
```

Comment:
we can remove the `create::` at the beginning of the `use` here.  We don't need to use the root of the create for this 👍